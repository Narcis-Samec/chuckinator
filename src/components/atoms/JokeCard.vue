<template>
  <v-card class="ma-2 pa-2 text-h4" height="250px">
    <v-chip
      variant="elevated"
      v-for="category in joke.categories"
      :key="joke.categories"
      color="primary"
      class="mb-2"
    >
      {{ category }}
    </v-chip>
    <p>{{ joke.value }}</p>
  </v-card>
</template>

<script>
import { mapState } from "vuex";

export default {
  computed: {
    ...mapState(["joke"]),
  },
};
</script>
