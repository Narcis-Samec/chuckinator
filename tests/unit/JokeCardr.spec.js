import { shallowMount } from '@vue/test-utils'
import JokeCard from "@/components/atoms/JokeCard";

describe('JokeCard.vue', () => {
  it('renders joke value correctly', () => {
    const joke = {
      categories: ['category1'],
      value: 'test joke',
    }
    const wrapper = shallowMount(JokeCard, {
      computed: {
        joke: () => joke
      }
    })
    const jokeValue = wrapper.find('p').text()
    expect(jokeValue).toBe('test joke')
  })

  it('matches snapshot', () => {
    const joke = {
      categories: ['category1'],
      value: 'test joke',
    }
    const wrapper = shallowMount(JokeCard, {
      computed: {
        joke: () => joke
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('does not truncate short jokes', () => {
    const joke = {
      categories: ['category1'],
      value: 'This is a short joke.'
    }
    const wrapper = shallowMount(JokeCard, {
      computed: {
        joke: () => joke
      }
    })
    const jokeText = wrapper.find('p').text()
    expect(jokeText).toBe('This is a short joke.')
  })
})