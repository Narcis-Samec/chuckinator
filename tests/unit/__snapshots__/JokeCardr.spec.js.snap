// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JokeCard.vue matches snapshot 1`] = `
<v-card class="ma-2 pa-2 text-h4" height="250px">
  <v-chip variant="elevated" color="primary" class="mb-2">category1</v-chip>
  <p>test joke</p>
</v-card>
`;
